docs/src/pages/components/data-grid/columns/columns.md

```diff
@@ -150,11 +150,11 @@ To disable column reordering, set the prop `disableColumnReorder={true}`.
 
 In addition, column reordering emits the following events that can be imported:
 
-- `COL_REORDER_START`: emitted when dragging of a header cell starts.
-- `COL_REORDER_DRAG_ENTER`: emitted when the cursor enters another header cell while dragging.
-- `COL_REORDER_DRAG_OVER`: emitted when dragging a header cell over another header cell.
-- `COL_REORDER_DRAG_OVER_HEADER`: emitted when dragging a header cell over the `ColumnsHeader` component.
-- `COL_REORDER_STOP`: emitted when dragging of a header cell stops.
+- `GRID_COLUMN_REORDER_START`: emitted when dragging of a header cell starts.
```

Comment:
We should discuss how to present events in the doc.
This is the constant name not its value. Would users use the constant or its value? 
I think we should show both, in a separate event page that we would link maybe at the bottom of this block

Reply:
+1 for:

- making the event const private
- only using the string values
- improve the types to only accept valid event strings

This way, we get the benefit of:

- Only one way to do things (we have to choose anyway what we encourage in the demos
- No pain of importing variables to subscribe to events (e.g. `document.addEventerListener('resize')`
- Retain type safety

Reply:
[DataTables](https://www.datatables.net/reference/event/) and [AG Grid](https://www.ag-grid.com/javascript-grid/grid-events/) only present the events as strings in their documentations.

[Firebase](https://firebase.google.com/docs/reference/js/firebase.analytics.EventName?hl=es) present the const and the value.

Reasons in favor of exporting event consts:
- Avoid hardcoded strings
- Refactoring is easier

Reply:
I guess I don't get how doing

```jsx
import { GRID_COLUMN_REORDER_START } from '@material-ui/x-grid';

apiRef.current.on(GRID_COLUMN_REORDER_START, () => {});
```

is better than:

```jsx
apiRef.current.on('gridColumnReoederStart', () => {});
```

It seems to yield the same features, without the import pain:

- if you did a typo, type safety get you covered
- if we release a breaking change, type safety get you covered
- if you want to refactor, the string is long enough to run search and replace 

I personally think that the `const` is only interesting for JavaScript users, that don't get type errors.

---

But I think that what's most important is the documentation. What do we use? For the demos, we can only use one variant. For the markdown, we could potentially describe the two but it would create confusion among developers (which one should I use ?!). So I think that we should first answer the following:

Assuming we only document one variant. Which one do we use in the documentation?

How about we only use the `const` at the cost of making it harder to use, but at the pros of making it better for JavaScript users? Developers that prefer string, can log the content of the constant, and we can explain them how the camelCase convention.

Reply:
> Assuming we only document one variant. Which one do we use in the documentation?

+1 for `eventName` in the demos and markdown pages. The only exception to show both would be in the [API page](https://material-ui.com/api/data-grid). There could have a "Events" section  with all strings and their equivalent constants, potentially showing the arguments like [Kendo UI](https://www.telerik.com/kendo-angular-ui/components/scheduler/api/SchedulerComponent/#toc-events-1) does. BTW they use `save` and `SaveEvent` in the demos because of the way Angular binds events.

I think something we're missing here is a stable naming convention for the events. Having `columnReordering:dragStart` and `columnHeaderClick` may create some confusion.

Reply:
@m4theushw  `eventName` over `EVENT_NAME`?

Should we do a vote? :)

- 🎉 `eventName`
- 🔥 `EVENT_NAME`